packages/sdk/src/types/runtime/actor.ts

```diff
@@ -80,20 +81,21 @@ export interface ActorSet {
  */
 export class Actor implements ActorLike, Patchable<ActorLike> {
     // tslint:disable:variable-name
-    private _internal: InternalActor;
+    private _internal = new InternalActor(this);
     /** @hidden */
     public get internal() { return this._internal; }
 
-    private _emitter: events.EventEmitter;
+    private _emitter = new events.EventEmitter();
     /** @hidden */
     public get emitter() { return this._emitter; }
 
     private _name: string;
     private _tag: string;
-    private _parentId: string;
+    private _exclusiveToUser: User;
+    private _parentId = ZeroGuid;
     private _subscriptions: SubscriptionType[] = [];
-    private _transform: ActorTransform;
-    private _appearance: Appearance;
+    private _transform = new ActorTransform();
+    private _appearance = new Appearance(this);

```

Comment:
>this [](start = 41, length = 4)

Is `this` accessible here? I thought it wasn't accessible until after super() was called in the ctor.

Reply:
I've never heard of that restriction, I don't know. It compiled, but I guess it could be null or something. I can check. I like initializing as much as possible in-line with the declaration though.

---
In reply to: [272676562](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/293#discussion_r272676562) [](ancestors = 272676562)